From 5d82ac6d7340c8034b0c61d248cde9442a398319 Mon Sep 17 00:00:00 2001
From: Bas Nijholt <bas@nijho.lt>
Date: Sun, 19 Nov 2023 15:06:41 -0800
Subject: [PATCH] Fix 'no instance of constructor
 "qsim::StateSpaceCUDA<FP>::Parameter::Parameter [with FP=float]" matches the
 argument list'

Full error:
./pybind_interface/cuda/pybind_main_cuda.cpp(30): error: no instance of constructor "qsim::StateSpaceCUDA<FP>::Parameter::Parameter [with FP=float]" matches the argument list
---
 pybind_interface/cuda/pybind_main_cuda.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/pybind_interface/cuda/pybind_main_cuda.cpp b/pybind_interface/cuda/pybind_main_cuda.cpp
index 57b0ba8..7480140 100644
--- a/pybind_interface/cuda/pybind_main_cuda.cpp
+++ b/pybind_interface/cuda/pybind_main_cuda.cpp
@@ -27,8 +27,10 @@ namespace qsim {
       unsigned num_sim_threads,
       unsigned num_state_threads,
       unsigned num_dblocks
-    ) : ss_params{num_state_threads, num_dblocks} {}
-
+    ) {
+      ss_params.num_threads = num_state_threads;
+      ss_params.num_dblocks = num_dblocks;
+    }
     StateSpace CreateStateSpace() const {
       return StateSpace(ss_params);
     }
-- 
2.39.2

